osu.Game/Screens/Play/GameplayClockExtensions.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Screens.Play
+{
+    public static class GameplayClockExtensions
+    {
+        /// <summary>
+        /// The rate of gameplay when playback is at 100%.
+        /// This excludes any seeking / user adjustments.
+        /// </summary>
+        public static double GetTrueGameplayRate(this IGameplayClock clock)
+        {
+            double rate = Math.Sign(clock.Rate);
+            foreach (double a in clock.GameplayAdjustments)
+                rate *= a;
+            return rate;
+        }
```

Comment:
If this sources the rate from the clock initially, then it's still going to be influenced by the pause frequency adjustment and the user playback rate, no? Shouldn't it be set to 1 initially then adjusted by the gameplay adjustments list?

Reply:
Did you happen to miss the `Math.Sign` call? I guess it could do with an inline comment, but this was to handle the case you mentioned of rewinding.

Reply:
Right, that makes sense. I was mostly concerned with the `Math.Sign(0) == 0` case but that should be resolved now.